pkg/controllers/provisioning/v1alpha1/allocation/controller.go

```diff
@@ -82,7 +82,7 @@ func (c *Controller) Reconcile(ctx context.Context, object client.Object) (recon
 		return reconcile.Result{}, fmt.Errorf("filtering pods, %w", err)
 	}
 	if len(pods) == 0 {
-		return reconcile.Result{}, nil
+		return reconcile.Result{RequeueAfter: c.Interval()}, nil
```

Comment:
What happens if you don't pass in the Interval? I just see that repeated a few times and wonder if in future people are going to forget to add it in the right cases?

Reply:
With this PR, we have two different types of controllers now, operating on strictly watch events, and strictly intervals. With this, we rely on each of the controllers to return their own result. For controllers operating on an interval, they need to return this in a reconcile.Result{} struct as you see above. 

For those forget to add an interval on a new controller, they won't see their controller working properly, so testing should fix this. It may be useful to add a note to help those who are looking at the code to understand it as well.